app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -218,9 +218,37 @@ class Settings private constructor(
 
     var shouldUseBottomToolbar by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_toolbar_bottom),
-        default = true
+        // Default accessibility users to top toolbar
+        default = !touchExplorationIsEnabled || !switchServiceIsEnabled
     )
 
+    /**
+     * Check each active accessibility service to see if it can perform gestures, if any can,
+     * then it is _likely_ a switch service is enabled. We are assuming this to be the case based on #7486
+     */
+    private val switchServiceIsEnabled: Boolean
```

Comment:
@eeejay does this seem like a good approach to checking for a switch service, then?

Reply:
Yeah, that looks good.